packages/protocol/contracts/L2/TaikoL2.sol

```diff
@@ -158,24 +141,41 @@ contract TaikoL2 is EssentialContract, TaikoL2Signer, ICrossChainSync {
         });
     }
 
+    function _update1559BaseFee(
+        uint64 l1Height,
+        uint32 gasInBlock
+    )
+        private
+        returns (uint256 baseFeePerGas)
+    {
+        if (gasExcess == 0) return 1;
+
+        (baseFeePerGas, gasExcess) = Lib1559Math.calcBaseFee({
+            numL1Blocks: latestSyncedL1Height == 0
+                ? 0
+                : l1Height - latestSyncedL1Height,
+            gasExcess: gasExcess,
+            gasInBlock: gasInBlock,
+            gasTarget: GAS_TARGET_PER_L1_BLOCK,
+            adjustmentQuotient: ADJUSTMENT_QUOTIENT
+        });
+        if (gasExcess == 0) gasExcess = 1;
```

Comment:
This one is just to prevent the storage slot to be reset to 0?

Reply:
Yes, here I use 0 to indicate the dynamic basefee calculate is disabled.